File: zone.js
import * as THREE from "https://cdn.skypack.dev/three";

export class Zone {
  constructor(scene) {
    this.radius = 80;

    this.mesh = new THREE.Mesh(
      new THREE.CircleGeometry(this.radius, 64),
      new THREE.MeshBasicMaterial({ color: 0x00ffff, wireframe: true })
    );
    this.mesh.rotation.x = -Math.PI / 2;
    scene.add(this.mesh);
  }

  update() {
    if (this.radius > 10) {
      this.radius -= 0.01;
      this.mesh.geometry.dispose();
      this.mesh.geometry = new THREE.CircleGeometry(this.radius, 64);
    }
  }

  isInside(pos) {
    return pos.length() < this.radius;
  }
}

File: zone.py
import math

class Zone:
    def __init__(self):
        self.x = 640
        self.y = 360
        self.radius = 300

    def update(self):
        if self.radius > 60:
            self.radius -= 0.05

    def is_inside(self, x, y):
        return math.hypot(x - self.x, y - self.y) < self.radius

    def draw(self, screen):
        import pygame
        pygame.draw.circle(
            screen,
            (0, 255, 255),
            (self.x, self.y),
            int(self.radius),
            2
        )
